server/public/shared/httpservice/httpservice.go

```diff
@@ -38,13 +43,25 @@ func splitFields(c rune) bool {
 	return unicode.IsSpace(c) || c == ','
 }
 
-func MakeHTTPService(configService configservice.ConfigService) HTTPService {
+func MakeHTTPService(configService getConfig) HTTPService {
 	return &HTTPServiceImpl{
 		configService,
 		RequestTimeout,
 	}
 }
 
+type pluginAPIConfigServiceAdapter struct {
+	pluginAPIConfigService plugin.API
+}
+
+func (p *pluginAPIConfigServiceAdapter) Config() *model.Config {
+	return p.pluginAPIConfigService.GetConfig()
+}
+
+func MakeHTTPServicePlugin(configService plugin.API) HTTPService {
+	return MakeHTTPService(&pluginAPIConfigServiceAdapter{configService})
+}
```

Comment:
You need some kind of wrapper because the config service is `Config()` but the plugin API provides `GetConfig()`. 
I guess it doesn't have to be implemented here, but its more convenient then duplicating it for every plugin that wants to use it.

Reply:
Cool, that makes sense.